api_app/api/routes/workspaces.py

```diff
@@ -133,6 +136,15 @@ async def delete_workspace(workspace=Depends(get_workspace_by_id_from_path), wor
 
     return WorkspaceIdInResponse(workspaceId=workspace.id)
 
+# workspace operations
+@workspaces_shared_router.get("/workspaces/{workspace_id}/operations", response_model=OperationInList, name=strings.API_GET_RESOURCE_OPERATIONS, dependencies=[Depends(get_current_workspace_owner_or_researcher_user_or_tre_admin)])
```

Comment:
Lets talk about what the appropriate user roles are for accessing these endpoints.

Reply:
So far i've copied the roles from the 'parent' resource, thinking being if you have permission to modify a resource then you can see the operations associated with it?

Reply:
I don't think a researcher can modify a workspace, or they shouldn't be able to anyway.

Reply:
you're right - a researcher user can _view_ (`GET`) a workspace. Do we want them to be able to see ops history too? If not then what should the access be for Workspace operations / WS Service Operations / User Resource Operations ?

Reply:
I think for now - the same people as can create/modify them - hence need to check the operations.

Reply:
this is done - only difference is the workspace operations, as the workspace PATCH was tre_admin only, but means that all the operations for workspaces would need to move to the core router which doesn't feel ideal - so i've gone with TRE Admin OR workspace owner who can see the ops.

Reply:
Make sense. The dependencies could be organized in a better way - https://github.com/microsoft/AzureTRE/issues/1075 .